src/InvertedIndices.jl

```diff
@@ -101,17 +101,49 @@ end
 InvertedIndexIterator(skips, picks) = InvertedIndexIterator{eltype(picks), typeof(skips), typeof(picks)}(skips, picks)
 Base.size(III::InvertedIndexIterator) = (length(III.picks) - length(III.skips),)
 
-@inline Base.iterate(I::InvertedIndexIterator) = iterate(I, (iterate(I.skips), iterate(I.picks)))
-Base.iterate(I::InvertedIndexIterator, ::Tuple{Any, Nothing}) = nothing
-@inline function Base.iterate(I::InvertedIndexIterator, (skipitr, pickitr))
+@inline function Base.iterate(I::InvertedIndexIterator)
+    skipitr = iterate(I.skips)
+    pickitr = iterate(I.picks)
+    pickitr === nothing && return nothing
+    while should_skip(skipitr, pickitr)
+        skipitr = iterate(I.skips, skipitr[2])
+        pickitr = iterate(I.picks, pickitr[2])
+        pickitr === nothing && return nothing
+    end
+    # This is a little silly, but splitting the tuple here allows inference to normalize
+    # Tuple{Union{Nothing, Tuple}, Tuple} to Union{Tuple{Nothing, Tuple}, Tuple{Tuple, Tuple}}
```

Comment:
Out of curiosity, have you noticed this across all supported versions or just in e.g. 1.11?

Reply:
The type stability unit tests fail without this manual union-split ternary, but pass with it.  It's the difference between putting a type-unstable value in a tuple v.s. returning two different tuples... and those fundamentals hold true going back to 1.0:

```
▶ julia +1.0 -q
julia> function f()
           x = rand([nothing, 1.0])
           return (x,)
       end
f (generic function with 1 method)

julia> @code_warntype f()
Body::Tuple{Union{Nothing, Float64}}
# ...

julia> function g()
           x = rand([nothing, 1.0])
           return x===nothing ? (nothing,) : (x,)
       end
g (generic function with 1 method)

julia> @code_warntype g()
Body::Union{Tuple{Nothing}, Tuple{Float64}}
```

Reply:
Oh interesting, I don't think I knew that. Thanks, I appreciate the explanation!